sentinel-adapter/sentinel-motan-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/motan/SentinelMotanProviderFilter.java

```diff
@@ -0,0 +1,73 @@
+package com.alibaba.csp.sentinel.adapter.motan;
+
+import com.alibaba.csp.sentinel.*;
+import com.alibaba.csp.sentinel.adapter.motan.config.MotanAdapterGlobalConfig;
+import com.alibaba.csp.sentinel.context.ContextUtil;
+import com.alibaba.csp.sentinel.log.RecordLog;
+import com.alibaba.csp.sentinel.slots.block.BlockException;
+import com.weibo.api.motan.common.MotanConstants;
+import com.weibo.api.motan.core.extension.Activation;
+import com.weibo.api.motan.core.extension.SpiMeta;
+import com.weibo.api.motan.exception.MotanAbstractException;
+import com.weibo.api.motan.rpc.Caller;
+import com.weibo.api.motan.rpc.Request;
+import com.weibo.api.motan.rpc.Response;
+
+/**
+ * program: sentinel-parent
+ * description: provider filter
+ * author: zhangxn8
+ **/
+@Activation(key =MotanConstants.NODE_TYPE_SERVICE)
+@SpiMeta(name = MotanAdapterGlobalConfig.SENTINEL_MOTAN_PROVIDER)
+public class SentinelMotanProviderFilter extends BaseMotanSentinelFilter {
+
+    public SentinelMotanProviderFilter(){
+        RecordLog.info("Sentinel motan provider filter initialized");
+    }
+
+    @Override
+    String getMethodName(Caller<?> caller, Request request, String prefix) {
+        return MotanUtils.getMethodResourceName(caller, request, prefix);
+    }
+
+    @Override
+    String getInterfaceName(Caller<?> caller, String prefix) {
+        return MotanUtils.getInterfaceName(caller, prefix);
+    }
+
+    @Override
+    public Response filter(Caller<?> caller, Request request) {
+        Entry interfaceEntry = null;
+        Entry methodEntry = null;
+        String prefix = MotanAdapterGlobalConfig.getMotanProviderPrefix();
+        String interfaceResourceName = getInterfaceName(caller, prefix);
+        String methodResourceName = getMethodName(caller, request, prefix);
+        try {
+            ContextUtil.enter(methodResourceName, "");
+            interfaceEntry = SphU.entry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN);
+            methodEntry = SphU.entry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN,
+                    request.getArguments());
+            Response result = caller.call(request);
```

Comment:
`ContextUtil.enter(methodResourceName, "");`  The second parameter is used to identify the caller, refer to `SentinelDubboProviderFilter`, which get the caller name by `DubboUtils#getApplication`. 
It seems that `MotanAppContextFilter` can be removed, since without it , we can also get the caller application from attachments, by calling `request.getAttachments().get("application")`.

Reply:
OK, Thank you